src/app/src/test/kotlin/org/imperial/mrc/hint/unit/ExtensionTests.kt

```diff
@@ -33,4 +39,31 @@ class ExtensionTests {
         res = Response(URL("http://whatever"), 500)
         assertThat(res.asResponseEntity().statusCode).isEqualTo(HttpStatus.INTERNAL_SERVER_ERROR)
     }
+
+    @Test
+    fun `asStreamingResponseEntity copies response to output stream`() {
+        val mockInputStream = mock<InputStream>{
```

Comment:
Could create a real stream here like 
` val inputStream = "test".byteInputStream()` and then at the end wrote back to a string to test it matches like 
```
  ByteArrayOutputStream().use {
            result.body?.writeTo(it)
            assertThat(it.toString()).isEqualTo("test")
        }
```
I'd find that easier to reason about than ` verify(mockOutputStream).write(any(), eq(0), eq(1))` - I'm not too sure what the latter really means!

Reply:
The code this is testing has all changed so test is now very different. I ended up putting the extension method on Request rather than DownloadRequest where it really belongs to make it possible to test with mockito (since DownloadRequest is a final class...)